test_opensearchpy/http_server.py

```diff
@@ -0,0 +1,53 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+#
+# Modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+import json
+import threading
+from http.server import BaseHTTPRequestHandler, HTTPServer
+
+hostName = "localhost"
+serverPort = 8080
+
+
+class Http_Server(BaseHTTPRequestHandler):
+    def do_GET(self):
+        self.send_response(200)
+        headers = self.headers
+        self.send_header("Content-type", "application/json")
+        self.end_headers()
+
+        Headers = {}
+        for header, value in headers.items():
+            capitalized_header = "-".join([word.title() for word in header.split("-")])
+            Headers.update({capitalized_header: value})
+        if "Connection" in Headers:
+            Headers.pop("Connection")
+
+        data = {"method": "GET", "headers": Headers}
+        self.wfile.write(json.dumps(data).encode("utf-8"))
+
+
+class MyHTTPServer(HTTPServer):
+    def __init__(self, host, port):
+        super().__init__((host, port), Http_Server)
+        self._server_thread = None
+
+    def start(self):
+        if self._server_thread is not None:
+            return
+
+        self._server_thread = threading.Thread(target=self.serve_forever)
+        self._server_thread.start()
+
+    def stop(self):
```

Comment:
This creates two `shutdown()` methods, which is weird. Overwrite `shutdown` or use an instance of `HTTPServer` instead of overriding it.

Reply:
while overwriting shutdown I am getting "recursionerror: maximum recursion depth exceeded while calling a python object". So I left it as stop method.